### ISSUE TITLE
[BUG] Type 'AngularPlugin' is not assignable to type 'ITelemetryPlugin'

### ISSUE THREAD

**Issue by erbsenkoenig** [microsoft/applicationinsights-angularplugin-js#51]
Having the following releases configured in the package.json
```
    "@microsoft/applicationinsights-angularplugin-js": "^2.8.1",
    "@microsoft/applicationinsights-web": "^2.8.0",
```

```
    "@microsoft/applicationinsights-angularplugin-js": "^2.8.0",
    "@microsoft/applicationinsights-web": "^2.8.0",
```

This configuration causes the application insights configuration to fail with the error:

```
this.applicationInsights = new ApplicationInsights({
      config: {
        enableRequestHeaderTracking: true,
        enableResponseHeaderTracking: true,
        enableCorsCorrelation: true,
        extensions: [this.angularPlugin], // THIS LINE DOESN'T COMPILE WITH BELOW ERROR
      },
    });
```

```
Type 'AngularPlugin' is not assignable to type 'ITelemetryPlugin'
Types of property 'setNextPlugin' are incompatible.
```

I had to fix the appinsights-web version to 2.7.4 to resolve the issue

**Comment by redmanmsc** [comment]
Same issue with the following error message:
 Type 'AngularPlugin' is not assignable to type 'ITelemetryPlugin'.
  Types of property 'setNextPlugin' are incompatible.
    Type '(next: ITelemetryPlugin | ITelemetryPluginChain) => void' is not assignable to type '(next: ITelemetryPluginChain | ITelemetryPlugin) => void'.
      Types of parameters 'next' and 'next' are incompatible.

**Comment by Tiguss** [comment]
Yep! Same issue here

```
Type 'AngularPlugin' is not assignable to type 'ITelemetryPlugin'.
  Types of property 'setNextPlugin' are incompatible.
```

**Comment by EdiWang** [comment]
Same here

![image](https://user-images.githubusercontent.com/3304703/164356270-3052b461-7ba5-4629-91cf-5497c2cda73f.png)

**Comment by MSNev** [comment]
v2.9.0 of the Angular plugin is now published, this is compatible with Applicaiton Insights ~2.8.1

I've also created issue #54 as the versions while we provide backward compatibility (all old API's should work with newer versions) we don't support forward compatibility older versions will work with newer components.

This is because the newer components may require dependencies which are only included in the newer based components.

I did just notice however, that the while the plugin package.json correctly has the ApplicationInsights set as "dependencies" the root package.json has them as devDependencies -- I'll create another issue for that so that it's explicitly identified.

But simplistically, we need to be more explicit

| AngularPlugin | Application Insights
|-----------------|-----------------------
| 2.9.0 | ~2.8.1 (will work with 2.8.0 but 2.8.0 has exception handling issue)
| 2.8.1 | ~2.7.4  (this was fixed with 2.8.1 to use ~2.7.4)
| 2.8.0 | ~2.7.x

**Comment by MSNev** [comment]
Oh, crap just realized about the additional complication with Angular 11 -> 13 upgrade....

Which does help the situation as 2.8.0 introduced Angular 13....

Sorry, the Angular plugin is not my primary focus and the person we had for this plugin is no longer with the team, so we are still back filling...

**Comment by MSNev** [comment]
I've just created an Angular11 branch that I'll try and fix some of these issues on we well.

**Comment by MSNev** [comment]
I should also add that the above are TypeScript typing errors / warnings the older Angular components will work with 2.8.1 (if you can get/bypass the mentions TS errors / warnings.

**Comment by MSNev** [comment]
Ok, I've just published v2.7.1 for Angular 11 which targets v2.7.4 of Application Insights.
I'm hoping this will resolve your TypeScript issues.

The reason for not targeting Application Insights v2.8.0 is that to do this properly it will need to implement the teardown capability to properly support the new `unload` functionality.

At runtime if using Application Insights v2.8.0 this version of the plugin will load and run, but you "may" still get TypeScript warnings if you attempt to compile them together.

**Comment by MSNev** [comment]
Closing for now, please comment if this does not resolve, I've also created a discussion Issue #57 to see how we can better handling versioning.
